doc/api/n-api.md

```diff
@@ -65,6 +67,41 @@ For example:
 #include <node_api.h>
 ```
 
+This will opt into the default `NAPI_VERSION` for the given release of Node.js.
+In order to ensure compatibility with specific versions of N-API, the version
+can be specified explicitly when including the header:
+
+```C
+#define NAPI_VERSION 3
+#include <node_api.h>
+```
+
+This restricts the N-API surface to just the functionality that was available in
+the specified (and earlier) versions.
+
+Some of the N-API surface is considered experimental and requires explicit
+opt-in to access those APIs:
+
+```C
+#define NAPI_EXPERIMENTAL
+#include <node_api.h>
+```
+
+In this case the entire API surface, including any experimental APIs, will be
+available to the module code.
+
+## N-API Version Matrix
+
+|       | 1       | 2        | 3        |
+|:-----:|:-------:|:--------:|:--------:|
+| v4.x  |         |          |          |
+| v6.x  |         |          | v6.14.2* |
+| v8.x  | v8.0.0* | v8.10.0* |          |
+| v9.x  | v9.0.0* | v9.3.0*  | v9.11.0* |
+| v10.x |         |          | v10.0.0  |
```

Comment:
As an aside do we need to update this table, at least in master?

Reply:
Yeah, I think we do.

Reply:
https://github.com/nodejs/node/issues/25801